clarify that `torch.finfo.tiny` is the smallest normal number

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63241

This is a common source of confusion, but it matches the NumPy
behavior.

Fixes #44010
Fixes #59526

Differential Revision: [D30307646](https://our.internmc.facebook.com/intern/diff/D30307646)

Commit:
clarify that `torch.finfo.tiny` is the smallest normal number

This is a common source of confusion, but it matches the NumPy
behavior.

Fixes #44010
Fixes #59526

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30307646).

Commit:
Update on "clarify that `torch.finfo.tiny` is the smallest normal number"


This is a common source of confusion, but it matches the NumPy
behavior.

Fixes #44010
Fixes #59526

Differential Revision: [D30307646](https://our.internmc.facebook.com/intern/diff/D30307646)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30307646).

mruberry commented on docs/source/type_info.rst:
```diff
@@ -26,13 +26,18 @@ bits        int     The number of bits occupied by the type.
 
+.. note::
+  `tiny` returns the smallest *normal* number, but there are smaller
+  subnormal numbers. See https://en.wikipedia.org/wiki/Denormal_number
```
"denormal numbers" would be consistent with the Wiki link

dagitses replied:
Yes but the floating point spec is what we truly want to be consistent with and it uses subnormal.

I can change the wiki link to https://en.wikipedia.org/wiki/Subnormal_number which redirects to the Denormal_number page. WDYT?

mruberry replied:
Probably not worth bothering

mruberry reviewed (approved):
Nice fix

dagitses commented:
> Nice fix

thanks!

Commit:
Update on "clarify that `torch.finfo.tiny` is the smallest normal number"


This is a common source of confusion, but it matches the NumPy
behavior.

Fixes #44010
Fixes #59526

Differential Revision: [D30307646](https://our.internmc.facebook.com/intern/diff/D30307646)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30307646).
